File: Minimum_Subarray.cpp
class Solution {
public:
    /**
     * @param nums: a list of integers
     * @return: A integer denote the sum of minimum subarray
     */
    int minSubArray(vector<int> nums) {
        // write your code here
        int s = 0;
        int result = INT_MAX;
        for (const auto& n : nums)
        {
            s += n;
            result = min(result, s);
            s = min(s, 0);
        }
        return result;
    }
};

File: Minimum Subarray.py
class Solution:
    """
    @param nums: a list of integers
    @return: A integer denote the sum of minimum subarray
    """
    def minSubArray(self, nums):
        # write your code here
        local = float('inf')
        gl = float('inf')
        for n in nums:
            local = min(local + n, n)
            gl = min(gl, local)
        
        return gl
